druid/examples/list.rs

```diff
@@ -34,31 +44,79 @@ fn main() {
 fn ui_builder() -> impl Widget<AppData> {
     let mut root = Flex::column();
 
+    // Build a button to add children to both lists
     root.add_child(
-        SizedBox::new(Button::new("Add", |_, data: &mut AppData, _| {
-            let mut d = (*data).deref().to_owned();
-            d.push((d.len() + 1) as u32);
-            *data = d.into();
-        }))
-        .height(30.0),
+        Button::new("Add", |_, data: &mut AppData, _| {
+            // Add child to left list
+            let value = data.left.len() + 1;
+            Arc::make_mut(&mut data.left).push(value as u32);
+
+            // Add child to right list
+            let value = data.right.len() + 1;
+            Arc::make_mut(&mut data.right).push(value as u32);
+        })
+        .fix_height(30.0),
         0.0,
     );
 
-    root.add_child(
+    let mut lists = Flex::row();
+
+    // Build a simple list
+    lists.add_child(
         Scroll::new(List::new(|| {
-            SizedBox::new(
-                Container::new(Padding::new(
-                    10.0,
-                    DynLabel::new(|d, _| format!("List item #{}", d)),
-                ))
-                .background(Color::rgb(0.5, 0.5, 0.5)),
-            )
-            .expand()
-            .height(50.0)
+            DynLabel::new(|item: &u32, _| format!("List item #{}", item))
+                .padding(10.0)
+                .expand()
+                .height(50.0)
+                .background(Color::rgb(0.5, 0.5, 0.5))
         }))
-        .vertical(),
+        .vertical()
+        .lens(AppData::left),
         1.0,
     );
 
+    // Build a list with shared data
+    lists.add_child(
+        Scroll::new(List::new(|| {
+            Flex::row()
+                .with_child(
+                    DynLabel::new(|(_, item): &(Arc<Vec<u32>>, u32), _| {
+                        format!("List item #{}", item)
+                    }),
+                    1.0,
+                )
+                .with_child(
+                    Button::sized(
+                        "Delete",
+                        |_ctx, (shared, item): &mut (Arc<Vec<u32>>, u32), _env| {
+                            // We have access to both child's data and shared data.
+                            // Remove element from right list.
+                            Arc::make_mut(shared).retain(|v| v != item);
+                        },
+                        80.0,
+                        20.0,
+                    ),
+                    0.0,
+                )
+                .padding(10.0)
+                .background(Color::rgb(0.5, 0.0, 0.5))
+                .fix_height(50.0)
+        }))
+        .vertical()
+        .lens(lens::Id.map(
+            // Expose shared data with children data
+            |d: &AppData| (d.right.clone(), d.right.clone()),
+            |d, x| {
```

Comment:
I think it might be helpful to write the types here for the reader. i.e. `d: &mut AppData, x: Arc<Vec<u32>>, Arc<Vec<u32>>` (are those the right types?)